common/chunkStatusLogger.go

```diff
@@ -326,6 +326,9 @@ func (csl *chunkStatusLogger) CloseLogger() {
 	 * On closing this channel the main logger will exit from its for-range loop.
 	 */
 	close(csl.unsavedEntries)
+
+	// Set address to nil so that GC will reclaim.
+	csl.unsavedEntries = nil
```

Comment:
Let's refrain from setting pointers to nil. This is not how references should be dropped and can confuse reader.